src/main/kotlin/com/atlassian/performance/tools/report/api/judge/Verdict.kt

```diff
@@ -14,20 +14,42 @@ class Verdict internal constructor(
     val reports: List<JUnitReport>,
     val failedActions: List<ActionType<*>>
 ) {
+    private companion object {
+        private val logger = LogManager.getLogger(this::class.java)
+    }
 
+    @Deprecated("Use Builder")
     constructor(
         reports: List<JUnitReport>
     ) : this(
         reports = reports,
         failedActions = emptyList()
     )
 
-    private val logger = LogManager.getLogger(this::class.java)
+    /**
```

Comment:
> JPERF-832 Add Verdict.Builder and mark constructor deprecated
> Now projects using this library can provide unit tests with Verdict constructed as those need

Do you have an example?

Reply:
In JPT we're using Verdict in SmartRerun and SmartRerunTest.
Previously there was full integration test producing Verdict and Junit reports using `report` lib logic. failedReports could be set only within the library because of internal constructor.
We relied on xml messages produced in this lib. In order to change it, we should be able to provide Verdict object with values we need.

Reply:
So this [internal PR](https://stash.atlassian.com/projects/JIRASERVER/repos/jira-performance-tests/pull-requests/977/overview) demonstrates the need for the API. Lemme see

Reply:
Ok, injecting `failedActions` leads to simpler test code, agreed :+1: 
A builder is a good idea, because `Verdict` started attracting new fields :+1:
Now I'll leave feedback on how to make the builder a bit more handy.